app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-search-tracker.js

```diff
@@ -0,0 +1,72 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+(function (Modules) {
+  'use strict'
+  // Tracks interactions with search forms, including on the all content finder ("site search"),
+  // homepage, and the layout super navigation header search.
+  class Ga4SearchTracker {
+    constructor ($module) {
+      this.$module = $module
+      this.$searchInput = this.$module.querySelector('input[type="search"]')
+
+      this.type = this.$module.dataset.ga4SearchType
+      this.url = this.$module.dataset.ga4SearchUrl
+      this.section = this.$module.dataset.ga4SearchSection
+      this.indexSection = this.$module.dataset.ga4SearchIndexSection
+      this.indexSectionCount = this.$module.dataset.ga4SearchIndexSectionCount
+    }
+
+    init () {
+      if (!this.$searchInput) {
+        console.warn('Ga4SearchTracker: Module added to element without child search input')
+        return
+      }
+
+      this.initialKeywords = this.$searchInput.value
+
+      if (window.GOVUK.getConsentCookie() && window.GOVUK.getConsentCookie().usage) {
+        this.start()
+      } else {
+        window.addEventListener('cookie-consent', () => this.start())
+      }
+    }
+
+    start () {
```

Comment:
Might be good to call this startTracking ?

I found myself wondering how this was different from init, which itself is confusing how it's different from the constructor 😆.

Reply:
Ah yes - actually `startModule` is the [canonical name](https://github.com/alphagov/govuk_publishing_components/blob/main/docs/javascript-modules.md#modules-and-cookie-consent) for modules that need cookie consent, so will update 👍